propertyestimator/layers/reweighting.py

```diff
@@ -42,24 +90,188 @@ def schedule_calculation(calculation_backend, storage_backend, layer_directory,
                                                 synchronous)
 
     @staticmethod
-    def perform_reweighting(physical_property, parameter_set, existing_data, **kwargs):
+    def perform_reweighting(physical_property, options, parameter_set_id, existing_data_paths,
+                            existing_force_field_paths, available_resources, **kwargs):
+
         """A placeholder method that would be used to attempt
         to reweight previous calculations to yield the desired
         property.
 
-        .. warning :: This method has not yet been implemented.
+        Warnings
+        --------
+        This method has not yet been implemented.
 
         Parameters
         ----------
-        physical_property: PhysicalProperty
+        physical_property: :obj:`propertyestimator.properties.PhysicalProperty`
             The physical property to attempt to estimate by reweighting.
-        parameter_set: ForceField
-            The force field parameters to use when estimating the property.
-        existing_data: list of StoredSimulationData
-            Data which has been stored from previous calculations on systems
-            of the same composition as the desired property.
+        parameter_set_id: :obj:`str`
+            The id of the force field parameters which the property should be
+            estimated with.
+        existing_data: :obj:`dict` of :obj:`str` and :obj:`str`
+            A file path to data which has been stored from previous calculations on
+            systems of the same composition as the desired property.
+        existing_force_fields: :obj:`dict` of :obj:`str` and :obj:`str`
+            A path to a dictionary of all of the force field parameters referenced by the
+            `existing_data`, which have been serialized with `serialize_force_field`
+        """
+        property_class = physical_property.__class__
+
+        if not hasattr(property_class, 'reweight'):
+            return None
+
+        existing_data = {}
+        existing_force_fields = {}
+
+        for substance_id in existing_data_paths:
+
+            existing_data[substance_id] = []
+
+            for data_path in existing_data_paths[substance_id]:
+
+                with open(data_path, 'rb') as file:
+                    existing_data[substance_id].append(pickle.load(file))
+
+        for force_field_id in existing_force_field_paths:
+
+            with open(existing_force_field_paths[force_field_id], 'rb') as file:
+                existing_force_fields[force_field_id] = deserialize_force_field(pickle.load(file))
+
+        reweighted_property = property_class.reweight(physical_property, options, parameter_set_id, existing_data,
```

Comment:
If `reweight` needs to access instance attribute, should it be an instance method rather than a `staticmethod`?